pkg/urbit/vere/pier.c

```diff
@@ -971,10 +971,22 @@ _pier_on_lord_slog(void* ptr_v, c3_w pri_w, u3_noun tan)
     c3_c* tan_c = u3r_string(tan);
     u3C.stderr_log_f(tan_c);
     c3_free(tan_c);
-    u3z(tan);
+
+    if ( 0 != pir_u->sog_f ) {
+      pir_u->sog_f(pir_u->sop_p, pri_w, tan);
+    }
+    else {
+      u3z(tan);
+    }
   }
   else {
-    u3_pier_tank(0, pri_w, tan);
+    u3_pier_tank(0, pri_w, u3k(tan));
+    if ( 0 != pir_u->sog_f ) {
+      pir_u->sog_f(pir_u->sop_p, pri_w, tan);
+    }
+    else {
+      u3z(tan);
+    }
```

Comment:
I'd refactor this slightly so that `u3z(tan)` is only called once, at the end of the function.